### PR TITLE
Minor cleanup & fix

### PR DESCRIPTION
**Pull request by naoyam**
Encountered while working on #2936

### CODE REVIEW

File: csrc/logical_domain_map.cpp [hunk 1]
```diff
@@ -693,11 +693,11 @@ bool ComputeAtLogicalDomainMap::canMap(
     const TensorDomain* td_b,
     const IterDomain* id_b) const {
   NVF_ERROR(
-      id_a->definition() == nullptr || id_a->isRFactorProduct(),
```
Comment (naoyam) [inline csrc/logical_domain_map.cpp:696]: Previously we assumed a root domain has no definition, which may not be the case anymore.

File: csrc/ir/internal_base_nodes.h [hunk 2]
```diff
@@ -529,22 +529,41 @@ class TensorDomain : public Val {
     return root_domain_.empty() ? logical_domain_ : root_domain_;
   };
 
+  bool isRoot(const IterDomain* id) const {
```
Comment (jacobhinkle) [inline csrc/ir/internal_base_nodes.h:532]: We might want to include a comment here indicating that this is always false if there is no root domain, i.e. this is not the same as checking `getMaybeRootDomain()` for `id`. Same goes for loop and allocation.

**Review by jacobhinkle — APPROVED** [review]
LGTM

**Comment by naoyam** [comment]
!build

**Pull request merged**